Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -460,12 +459,53 @@ protected class ClassWriter : IClassWriter {
       if (!createBody) { return null; }
       if (isStatic || NeedsCustomReceiver(member)) { wr.WriteLine("@staticmethod"); }
       wr.Write($"def {name}(");
-      WriteFormals(formals, isStatic && !NeedsCustomReceiver(member), wr);
+      var sep = "";
+      WriteRuntimeTypeDescriptorsFormals(member, ForTypeDescriptors(typeArgs, member, lookasideBody), wr, ref sep, tp => tp.CompileName);
+      WriteFormals(formals, isStatic && !NeedsCustomReceiver(member), ref sep, wr);
       return wr.NewBlockPy("):", close: BlockStyle.Newline);
     }
 
     protected override string TypeDescriptor(Type type, ConcreteSyntaxTree wr, IToken tok) {
-      throw new UnsupportedFeatureException(Token.NoToken, Feature.RuntimeTypeDescriptors);
+      Contract.Requires(type != null);
+      Contract.Requires(tok != null);
+      Contract.Requires(wr != null);
+
+      var customName = false;
+      var name = type.NormalizeExpandKeepConstraints() switch {
+        // unresolved proxy; just treat as bool, since no particular type information is apparently needed for this type
+        BoolType or TypeProxy => "bool",
+        IntType or BitvectorType => "int",
+        RealType => "real",
+        UserDefinedType udt => udt.ResolvedClass switch {
+          TypeParameter tp => TypeParameterDescriptor(tp),
+          ClassDecl => "null",
+          TupleTypeDecl tpl => TupleTypeDescriptor(tpl, udt.TypeArgs, udt.tok),
+          NewtypeDecl => NewtypeDescriptor(udt),
+          _ => throw new cce.UnreachableException()
+        },
+        _ => throw new cce.UnreachableException()
+      };
+      return (customName ? "" : DafnyDefaults + ".") + name;
```

Comment:
I think this is a bit too confusing to be worth it. Consider just duplicating the reference inline instead.